File: src/main/java/com/DavidM1A2/AfraidOfTheDark/common/block/gravewood/BlockGravewoodLeaves.java
/*
 * Author: David Slovikosky
 * Mod: Afraid of the Dark
 * Ideas and Textures: Michael Albertson
 */
package com.DavidM1A2.AfraidOfTheDark.common.block.gravewood;

import java.util.Random;

import com.DavidM1A2.AfraidOfTheDark.common.block.core.AOTDLeaves;
import com.DavidM1A2.AfraidOfTheDark.common.initializeMod.ModBlocks;

import net.minecraft.block.state.IBlockState;
import net.minecraft.item.Item;

public class BlockGravewoodLeaves extends AOTDLeaves
{
	/*
	 * Define gravewood leaves block
	 */
	public BlockGravewoodLeaves()
	{
		super();
		this.setUnlocalizedName("gravewoodLeaves");
		this.setRegistryName("gravewoodLeaves");
	}

	@Override
	public Item getItemDropped(IBlockState state, Random rand, int fortune)
	{
		return Item.getItemFromBlock(ModBlocks.gravewoodSapling);
	}
}

File: src/main/kotlin/com/davidm1a2/afraidofthedark/common/block/gravewood/BlockGravewoodLeaves.kt
package com.davidm1a2.afraidofthedark.common.block.gravewood

import com.davidm1a2.afraidofthedark.common.block.core.AOTDBlockLeaves
import com.davidm1a2.afraidofthedark.common.constants.ModBlocks
import net.minecraft.block.material.Material
import net.minecraft.block.state.IBlockState
import net.minecraft.util.IItemProvider
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World

/**
 * Gravewood leaves are grown by gravewood trees
 *
 * @constructor just sets the name of the block
 */
class BlockGravewoodLeaves : AOTDBlockLeaves("gravewood_leaves", Properties.create(Material.LEAVES)) {
    override fun getItemDropped(state: IBlockState, world: World, blockPos: BlockPos, fortune: Int): IItemProvider {
        return ModBlocks.GRAVEWOOD_SAPLING
    }
}